src/appleseed-max-impl/utilities.cpp

```diff
@@ -153,6 +153,25 @@ void update_map_buttons(IParamMap2* param_map)
     }
 }
 
+float get_output_amount(Texmap* texmap, TimeValue t)
+{
+    float output_value = 1.0f;
+    for (int i = 0, e = texmap->NumRefs(); i < e; ++i)
+    {
+        ReferenceTarget* ref = texmap->GetReference(i);
+        if (ref != nullptr && ref->SuperClassID() == TEXOUTPUT_CLASS_ID)
+        {
+            StdTexoutGen* tex_output = dynamic_cast<StdTexoutGen*>(ref);
+            if (tex_output != nullptr)
+            {
+                tex_output->Update(t, FOREVER);
+                output_value = tex_output->GetOutAmt(t);
```

Comment:
Just to be sure: this function returns the output amount of the _last_ `StdTexoutGen`, right? I.e. `output_value` gets repeatedly overwritten.